src/sinks/humio_logs.rs

```diff
@@ -36,29 +39,26 @@ pub enum Encoding {
     Json,
 }
 
-impl Into<splunk_hec::Encoding> for Encoding {
-    fn into(self) -> splunk_hec::Encoding {
-        splunk_hec::Encoding::Json
+impl From<Encoding> for splunk_hec::Encoding {
+    fn from(v: Encoding) -> Self {
+        match v {
+            Encoding::Json => splunk_hec::Encoding::Json,
+        }
     }
 }
 
 #[typetag::serde(name = "humio_logs")]
 impl SinkConfig for HumioLogsConfig {
     fn build(&self, cx: SinkContext) -> crate::Result<(super::RouterSink, super::Healthcheck)> {
-        if self.encoding.codec != Encoding::Json {
+        if self.encoding.codec() != &Encoding::Json {
             error!("Using an unsupported encoding for Humio");
         }
```

Comment:
I know this isn't directly relevant to this change set, but is this a useful test to be doing? Serde should have already ensured that the encoding can only be one of the valid variants in the enum.
 
I'm also a bit surprised Rust doesn't warn about this comparison being unconditionally false. Probably just a bit beyond its const checking.

Reply:
Absolutely.